src/SlackEventHandler.ts

```diff
@@ -325,4 +341,24 @@ export class SlackEventHandler extends BaseSlackHandler {
         await room.onSlackTyping(typingEvent, teamId);
     }
 
+    private async handleTeamSyncEvent(event: ISlackEvent, teamId: string) {
+        if (!this.main.teamSyncer) {
+            throw Error("ignored");
+        }
+        if (event.type === "channel_created") {
+            // Note: Slack violates the usual stringness of 'channel' here.
+            const eventDetails = event as unknown as ISlackChannelAdded;
+            await this.main.teamSyncer.onChannelAdded(teamId, eventDetails.channel.id, eventDetails.channel.name, eventDetails.channel.creator);
+        } else if (event.type === "channel_deleted") {
+            await this.main.teamSyncer.onChannelDeleted(teamId, event.channel);
+        } else if (event.type === "team_join") {
+            const user = event.user as unknown as ISlackUser;
```

Comment:
What type is `event.type` that we need to use double assertion?

Reply:
Good catch.